interface/connection/conn.go

```diff
@@ -4,55 +4,97 @@
 package connection
 
 import (
+	"fmt"
 	"io"
 	"net"
+	"os"
 	"sync"
 )
 
+const defaultUnixSocketAddress = "/run/parsec/parsec.sock"
+const parsecEndpointEnvironmentVariable = "PARSEC_SERVICE_ENDPOINT"
+
+// Connection represents a connection to the parsec service.
 type Connection interface {
+	// Open should be called before use
 	Open() error
+	// Methods to read, write and close the connection
 	io.ReadWriteCloser
 }
 
+// type to manage unix socket connection
 type unixConnection struct {
 	sync.Mutex
```

Comment:
Why is the mutex needed here? Is this structure meant to be accessed from multiple threads?
If so, could the open and close calls be made locally on each read and write so that you would probably not need the mutex and the `isOpen` bool? I think we do like this on the Rust client. I assume the overhead would not be too big for that.

Reply:
have remove this - without thinking i'd tried to make thread safe, but it is opened and closed for each operation call (doing it on each read/write breaks the unix socket connection.  have added comment about thread safety